zerver/lib/export.py

```diff
@@ -1280,51 +1280,50 @@ def get_ids(records: List[Record]) -> Set[int]:
         messages_we_sent_to_them = Message.objects.filter(
             sender__in=consented_user_ids,
             recipient__in=recipient_ids_for_them,
-        ).order_by("id")
+        )
 
         message_queries = [
             messages_we_received,
             messages_we_sent_to_them,
         ]
 
     all_message_ids: Set[int] = set()
-    dump_file_id = 1
 
     for message_query in message_queries:
-        dump_file_id = write_message_partial_for_query(
-            realm=realm,
-            message_query=message_query,
-            dump_file_id=dump_file_id,
-            all_message_ids=all_message_ids,
-            output_dir=output_dir,
-            user_profile_ids=user_ids_for_us,
-            chunk_size=chunk_size,
-        )
+        message_ids = set(get_id_list_gently_from_database(base_query=message_query, id_field="id"))
+
+        # We expect our queries to be disjoint, although this assertion
+        # isn't stricly necessary if you don't mind a little bit of
+        # overhead.
+        assert len(message_ids.intersection(all_message_ids)) == 0
+
+        all_message_ids |= message_ids
+
+    message_id_chunks = chunkify(sorted(list(all_message_ids)), chunk_size=MESSAGE_BATCH_CHUNK_SIZE)
+
+    write_message_partials(
+        realm=realm,
+        message_id_chunks=message_id_chunks,
+        output_dir=output_dir,
+        user_profile_ids=user_ids_for_us,
+    )
 
     return all_message_ids
 
 
-def write_message_partial_for_query(
+def write_message_partials(
+    *,
     realm: Realm,
-    message_query: Any,
-    dump_file_id: int,
-    all_message_ids: Set[int],
+    message_id_chunks: List[List[int]],
     output_dir: Path,
     user_profile_ids: Set[int],
-    chunk_size: int = MESSAGE_BATCH_CHUNK_SIZE,
-) -> int:
-    min_id = -1
-
-    while True:
-        actual_query = message_query.filter(id__gt=min_id)[0:chunk_size]
-        message_chunk = make_raw(actual_query)
-        message_ids = {m["id"] for m in message_chunk}
-        assert len(message_ids.intersection(all_message_ids)) == 0
+) -> None:
 
-        all_message_ids.update(message_ids)
+    dump_file_id = 1
 
-        if len(message_chunk) == 0:
-            break
+    for message_id_chunk in message_id_chunks:
+        actual_query = Message.objects.filter(id__in=message_id_chunk)
```

Comment:
This should have an `order_by`, right?

Reply:
Fixed.  (It's technically unnecessary due to downstream sorting, but harmless to add.)